### ISSUE TITLE
SMT sequential inserts fail

### ISSUE THREAD

**Issue by frisitano** [0xPolygonMiden/miden-vm#1041]
I'm having some problems executing multiple inserts in a row against a SMT.  I receive an error of the following nature:

```
---- collections::smt::tsmt_insert_16 stdout ----
thread 'collections::smt::tsmt_insert_16' panicked at 'called `Result::unwrap()` on an `Err` value: AdviceKeyNotFound([BaseElement(15556198973653036125), BaseElement(2569263626647358186), BaseElement(12115666839279463362), BaseElement(5570871396401658231)])', test-utils/src/lib.rs:289:36
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- collections::smt::tsmt_insert_32 stdout ----
thread 'collections::smt::tsmt_insert_32' panicked at 'called `Result::unwrap()` on an `Err` value: AdviceKeyNotFound([BaseElement(6914814850777475923), BaseElement(1672658896067019021), BaseElement(11491567203765900427), BaseElement(8641048661315345931)])', test-utils/src/lib.rs:289:36

---- collections::smt::tsmt_insert_48 stdout ----
thread 'collections::smt::tsmt_insert_48' panicked at 'called `Result::unwrap()` on an `Err` value: AdviceKeyNotFound([BaseElement(6110365337961474275), BaseElement(16137260194458567299), BaseElement(3850080979430617408), BaseElement(7925048865637296829)])', test-utils/src/lib.rs:289:36
```

To recreate, replace the test brace [here](https://github.com/0xPolygonMiden/miden-vm/blob/57110fa60c27e64cfdf3c8a6fd8cb4a39a79baf2/stdlib/tests/collections/smt.rs#L204-L218) with:

```rust
fn assert_insert(
    init_smt: &TieredSmt,
    key: RpoDigest,
    old_value: Word,
    new_value: Word,
    new_root: RpoDigest,
) {
    let old_root = init_smt.root();
    let source = r#"
        use.std::collections::smt

        begin
            dupw.1 movdnw.3
            exec.smt::insert

            dropw swapw
            push.1.0.0.1

            exec.smt::insert
        end
    "#;
```

Apologies for lack of a better example.

**Comment by bobbinth** [comment]
Ah! I think I know what's happening. The current code does not insert new leaf pre-images into the advice provider. I'll fix it today.

**Comment by bobbinth** [comment]
Closed by #1042.
